### ISSUE TITLE
EXEC for aborted transaction returns scalar null reply instead of array null reply

### ISSUE THREAD

**Issue by ashanbrown** [alicebob/miniredis#200]
I manually ran through the test at https://github.com/alicebob/miniredis/blob/cd3acaf48b849fb1c9c42c10a7cc533bb8550997/cmd_transactions_test.go#L207 against my local redis instance (version 5.0.7) and the response code on a transaction failure is different than the test.

Here's the trace of client 1 in that test:

```
SET one three
+OK
WATCH one
+OK
MULTI
+OK
GET one
+QUEUED
EXEC
*-1
````

According to https://redis.io/commands/exec,
> When using WATCH, EXEC can return a Null reply if the execution was aborted. ,

`EXEC` returns an array, so we should expect an array null reply, which according to https://redis.io/topics/protocol#nil-reply, is `*-1`, so the redis server is behaving as expected.

I think the test should be calling `mustNilList` rather than `mustNil`.

**Comment by alicebob** [comment]
> I think the test should be calling `mustNilList` rather than `mustNil`.

Thanks for the detailed description. I'll have a look soon, or feel free to
make a PR :)

**Comment by alicebob** [comment]
That's a good find! Can you test if branch `execnil` fixes it (#201)?

**Comment by ashanbrown** [comment]
@alicebob That change works for me, thanks.
